### ISSUE TITLE
Restoring MBR backups can lead to inability to boot (due to loose file pattern match)

### ISSUE THREAD

**Issue by shasheene** [rescuezilla/rescuezilla#162]
A user on the Sourceforge forum has [reported](https://sourceforge.net/p/rescuezilla/discussion/support/thread/3397566442/#2999) that restoring a Windows 10 image over the network causes his machine not too boot, **but restoring the same image over a locally attached USB works fine**. After examining his log files the reason became clear. Rescuezilla is overwriting the Master Boot Record with the wrong file.

The locally attached USB drive case:

> Restoring the first 446 bytes of MBR data (executable code area) for /dev/sda. Running: dd if=/mnt/backup/2020-12-30-rescuezilla-LM-straight-to-USB/sda-mbr of=/dev/sda bs=446 
> Restoring the first 446 bytes of MBR data (executable code area) for /dev/sda: dd if=/mnt/backup/2020-12-30-rescuezilla-LM-straight-to-USB/sda-mbr of=/dev/sda bs=446  returned 0:  1+1 records in
> 1+1 records out
> 512 bytes copied, 0.000925328 s, 553 kB/s

Compared to the network shared folder case:

> Restoring the first 446 bytes of MBR data (executable code area) for /dev/sda. Running: dd if=/mnt/backup/2020-12-30-img-rescuezilla/sda-hidden-data-after-mbr of=/dev/sda bs=446 
> Restoring the first 446 bytes of MBR data (executable code area) for /dev/sda: dd if=/mnt/backup/2020-12-30-img-rescuezilla/sda-hidden-data-after-mbr of=/dev/sda bs=446  returned 0:  2349+1 records in
> 2349+1 records out
> 1048064 bytes (1.0 MB, 1.0 MiB) copied, 0.161952 s, 6.5 MB/s

The root cause is appears to be the pattern matching code is matching both the `sda-mbr` and the `sda-hidden-data-after-mbr` files, when it should only be matching the `sda-mbr` file:
```
            # There is a maximum of 1 MBR per drive (there can be many drives). Master Boot Record (MBR) is never
            # listed in 'parts' list.
            mbr_glob_list = glob.glob(os.path.join(dir, short_disk_device_node) + "*-mbr")
```

**This issue will affect any backup of an MBR disk created with either Clonezilla or Rescuezilla v2.0-v2.1.1.** The contents of the backup are completely fine, it's just a bug in the logic of a restore operation affecting Rescuezilla v2.0-v2.1.1 that's causing the issue.

I still haven't managed to reproduce this issue: it doesn't occur when restoring from a locally attached disks, and hadn't occurred when I've done a restore from a Linux-based SMB/CIFS network share. I have just tested a restore from a Windows 10 shared folder and the pattern matching selected the `sda-mbr` file so worked as intended. It's unclear to me why the pattern match using the Python "glob" function apparently produces inconsistent results (it could be due to the underlying filesystem, timestamp or something else). **But regardless, it's clear where the root cause of the issue is. I will fix the logic and release Rescuezilla v2.1.2.**

**Comment by 45bcdf** [comment]
Hi Shasheene, I'm the one (richard) you are working with over on SourceForge about this issue. I just wanted to clarify that my restore issues are happening with a Linux Mint restore image using a Linux-based SMC/CIFS (Open Media Vault 5 server running on top of Raspian). My log files I sent reflect this Linux to Linux restore environment. One final thing to note (speaking of  Windows), the usb drive that I have used to successfully restore my images is an NTFS drive vs. ext4 drives connected to my NAS. Might be nothing but I thought I would share those exra details.

**Comment by shasheene** [comment]
Thanks for the information.

Regardless of why the variation is happening, the root cause is an incorrect asterisk wildcard character in the pattern match `glob.glob(os.path.join(dir, short_disk_device_node) + "*-mbr")`, so it's a very simple fix.

I introduced the bug when implementing the EBR (Extended Boot Record) case for Rescuezilla v2.0: the eg, `sda4-ebr` file _does_ need the asterisk (because the EBR is associated with partition 4), but for the MBR case the filename is always eg, `sda-mbr` (the MBR is associated with the device but not any partitions), so the wildcard match asterisk then causes the unintended effect.

**Comment by shasheene** [comment]
@45bcdf 

**Can you please try the following image and confirm it resolves your issue:** https://drive.google.com/file/d/1q116hzKibOEoH35oTc059Vw4Lt_HEDDS/view?usp=sharing

Here are the checksums if you want them:

```
sha256sum:
8330af6e2bd2e53100d2f08d19f6ac7345224760154401a1cd121d6e4d253326 rescuezilla-2.1.2-64bit.groovy.iso
934b290196ebbd8ae47bde56d41c5d11bfadd7812e39b16e2807501e6a1ca053 rescuezilla-2.1.2-64bit.focal.iso

sha1sum:
10467ec9b75ab7ecbc91eccdb3bee4cb27df0fdc rescuezilla-2.1.2-64bit.groovy.iso
bcd031e1a42ceb8c4cf9df8da55bcc8116e9518c rescuezilla-2.1.2-64bit.focal.iso

md5sum:
d8c128a06f3a817e8c9b932d127465ba rescuezilla-2.1.2-64bit.groovy.iso
9b1a063642f79d596ee4aa125400dca6 rescuezilla-2.1.2-64bit.focal.iso
```

I will likely be releasing making a new release in about 14 hours time.

**Comment by shasheene** [comment]
OK. @45bcdf 

I have released [Rescuezilla v2.1.2](https://github.com/rescuezilla/rescuezilla/releases/latest) which fixes the incorrect filename pattern match that was causing the issue.

Please download that version and let me know how you go.

(Please note Richard, if you downloaded the test version I uploaded yesterday, please note it based on a git commit that has been rebased so the official v2.1.2 is a rebuild and therefore has a different checksum.)

For completeness I note here that I did notice that Rescuezilla's command to restore the MBR is restoring the full file, rather than having "count=1" to restore 446 bytes as the comment is indicating. Restoring the executable section of the MBR is no big deal and I don't want to make unnecessary changes in a patch update. However as part of moving Rescuezilla closer to Clonezilla in task #146, I will be auditing the entire Clonezilla implementation and completing task #151 to add an automated testing suite for Rescuezilla's long-term maintainability,
